base_location/models/res_partner.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2016 Nicolas Bessi, Camptocamp SA
+# Copyright 2018 Tecnativa - Pedro M. Baeza
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    zip_id = fields.Many2one('res.city.zip', 'ZIP Location')
+
+    @api.onchange('city_id')
+    def _onchange_city_id(self):
+        if not self.zip_id:
+            super(ResPartner, self)._onchange_city_id()
+        if self.zip_id and self.city_id != self.zip_id.city_id:
+            self.zip_id = False
```

Comment:
```python
self.update({
    'zip_id': False,
    'zip': False,
    'city': False,
})
```

Reply:
Does this have an impact in the performance?